producers/src/main/java/uk/gov/moj/sdt/producers/sdtws/SdtInternalEndpointPortType.java

```diff
@@ -1,135 +1,142 @@
+    /**
+     * Logger object.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdtInternalEndpointPortType.class);
+
+    /**
+     * Update item handler for handling individual request update.
+     */
+    private IWsUpdateItemHandler updateItemHandler;
+
+    @Autowired
+    public SdtInternalEndpointPortType(@Qualifier("WsUpdateItemHandler") IWsUpdateItemHandler updateItemHandler) {
+        setUpdateItemHandler(updateItemHandler);
+    }
+
+    @Override
+    public UpdateResponseType updateItem(final UpdateRequestType updateRequest) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Endpoint called for update item for request reference [" +
+                    updateRequest.getHeader().getSdtRequestId() + "]");
+        }
+
+        if (PerformanceLogger.isPerformanceEnabled(PerformanceLogger.LOGGING_POINT_2)) {
+            final StringBuffer detail = new StringBuffer();
+            detail.append("\n\n\tsdt sdt request id=" + updateRequest.getHeader().getSdtRequestId() +
+                    "\n\tstatus code=" + updateRequest.getStatus().getCode() + "\n");
+
+            if (updateRequest.getStatus().getError() != null) {
+                detail.append("\n\terror code=" + updateRequest.getStatus().getError().getCode() +
+                        "\n\terror description=" + updateRequest.getStatus().getError().getDescription() + "\n");
+            }
+
+            // Write message to 'performance.log' for this logging point.
+            PerformanceLogger.log(this.getClass(), PerformanceLogger.LOGGING_POINT_2, "Status update received",
+                    detail.toString());
+        }
+
+        UpdateResponseType response = null;
+
+        try {
+            response = updateItemHandler.updateItem(updateRequest);
+        }
+        catch (Exception throwable)
+        {
+            handleThrowable(throwable);
+        }
+
+        if (PerformanceLogger.isPerformanceEnabled(PerformanceLogger.LOGGING_POINT_9)) {
+            final StringBuffer detail = new StringBuffer();
+            detail.append("\n\n\tstatus code=" + response.getStatus().getCode() + "\n");
+
+            if (response.getStatus().getError() != null) {
+                detail.append("\n\terror code=" + response.getStatus().getError().getCode() +
+                        "\n\terror description=" + response.getStatus().getError().getDescription() + "\n");
+            }
+
+            // Write message to 'performance.log' for this logging point.
+            PerformanceLogger.log(this.getClass(), PerformanceLogger.LOGGING_POINT_9,
+                    "Status update response returned", detail.toString());
+        }
+
+        return response;
+    }
+
+    /**
+     * Handles throwable and re-throws runtime exception.
+     *
+     * @param throwable exception to be handled
+     */
+    private void handleThrowable(final Throwable throwable) {
```

Comment:
Should this be renamed as handleException() now that it's being called with an Exception instead of Throwable?  If so then don't forget to change parameter type and name and JavaDoc comment.